src/device_type.c

```diff
@@ -206,3 +206,37 @@ hash_device_type(PG_FUNCTION_ARGS)
 {
   return hash_uint32((int32) PG_GETARG_UINT8(0));
 }
+
+PG_FUNCTION_INFO_V1(int4_to_device_type);
+Datum int4_to_device_type(PG_FUNCTION_ARGS)
+{
+	device_type c = PG_GETARG_INT32(0);
+
+    /* XXX we do not check input validity for performance reasons */
```

Comment:
I don't think this is a good idea as it risk that invalid values get stored on disk which makes reading back impossible